src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3800,30 +3764,61 @@ public Object stringCharacterIndex(DynamicObject string, DynamicObject pattern,
         }
     }
 
+    // Named 'string_byte_index' in Rubinius.
     @Primitive(name = "string_byte_index_from_char_index", needsSelf = false, lowerFixnum = 2)
-    @ImportStatic(StringGuards.class)
+    @ImportStatic({ StringGuards.class, StringOperations.class })
     public static abstract class StringByteIndexFromCharIndexNode extends PrimitiveArrayArgumentsNode {
 
         public abstract Object executeFindByteIndex(DynamicObject string, int characterIndex);
 
-        @Specialization(guards = "characterIndex < 0")
-        protected Object byteIndexNegativeIndex(DynamicObject string, int characterIndex) {
-            throw new RaiseException(
-                    getContext().getCoreExceptions().argumentError(
-                            coreStrings().CHARACTER_INDEX_NEGATIVE.getRope(), this));
-        }
-
-        @Specialization(guards = "characterIndexTooLarge(string, characterIndex)")
-        protected Object byteIndexTooLarge(DynamicObject string, int characterIndex) {
-            return nil();
+        @Specialization(guards = "!characterIndexInBounds(string, characterIndex)")
+        protected Object indexOutOfBounds(DynamicObject string, int characterIndex,
+                @Cached("createBinaryProfile()") ConditionProfile negativeIndexProfile) {
+            if (negativeIndexProfile.profile(characterIndex < 0)) {
+                throw new RaiseException(
+                        getContext().getCoreExceptions().argumentError(
```

Comment:
FWIW there is a `coreExceptions()` helper which would allow to not split this in 3 lines but 2.

Reply:
Thanks. This was old code, but I'll update it.